lib/android-tools-info.ts

```diff
@@ -0,0 +1,173 @@
+		return ((): number => {
+			if(!this.selectedCompileSdk) {
+				let latestValidAndroidTarget = this.getLatestValidAndroidTarget().wait();
+				if(latestValidAndroidTarget) {
+					let integerVersion = this.parseAndroidSdkString(latestValidAndroidTarget);
+
+					if(integerVersion && integerVersion >= AndroidToolsInfo.MIN_REQUIRED_COMPILE_TARGET) {
+						return integerVersion;
+					}
+				}
+
+				this.printMessage(`Could not find supported Android target for compilation. Please install at least: ${AndroidToolsInfo.MIN_REQUIRED_COMPILE_TARGET}.` +
+							" Make sure you have the latest Android tools installed as well.",
+							'Run "android" from your command-line to install required Android SDK version.');
+			}
+
+			return this.selectedCompileSdk;
+		}).future<number>()();
+	}
+
+	private getTargetSdk(): IFuture<number> {
+		return ((): number => {
+			let targetSdk = this.$options.sdk ? parseInt(this.$options.sdk) : this.getCompileSdk().wait();
+			let newTarget = `${AndroidToolsInfo.ANDROID_TARGET_PREFIX}-${targetSdk}`;
+			if(!_.contains(AndroidToolsInfo.SUPPORTED_TARGETS, newTarget)) {
+				let supportedVersions = AndroidToolsInfo.SUPPORTED_TARGETS.sort();
+				let minSupportedVersion = this.parseAndroidSdkString(_.first(supportedVersions));
+				let maxSupportedVersion = this.parseAndroidSdkString(_.last(supportedVersions));
+				if(targetSdk && (targetSdk < minSupportedVersion)) {
+					this.printMessage(`The selected target SDK ${newTarget} is not supported. You should target at least ${minSupportedVersion}.`);
+				} else if(!targetSdk || targetSdk > maxSupportedVersion) {
+					this.$logger.warn(`The selected Android target '${newTarget}' is not verified as supported. Some functionality may not work as expected.`);
+				}
+			}
+
+			return targetSdk;
+		}).future<number>()();
+	}
+
+	private getBuildToolsVersion(): IFuture<string> {
+		return ((): string => {
+			let pathToBuildTools = path.join(this.getAndroidHome(), "build-tools");
+			let maxSupportedAndroidTarget = this.parseAndroidSdkString(_.last(AndroidToolsInfo.SUPPORTED_TARGETS));
+			let buildToolsRange = `${AndroidToolsInfo.REQUIRED_BUILD_TOOLS_RANGE_PREFIX} <=${maxSupportedAndroidTarget}`;
+			if(this.$fs.exists(pathToBuildTools).wait()) {
+				let availableBuildToolsVersions = this.$fs.readDirectory(pathToBuildTools).wait()
+					.filter(buildTools => !!buildTools.match(/^(\d+\.){2}\d+$/));
+				let selectedBuildToolsVersion = semver.maxSatisfying(availableBuildToolsVersions, buildToolsRange);
+				if(selectedBuildToolsVersion) {
+					return selectedBuildToolsVersion;
+				}
+				this.$logger.trace(`Installed build tools are ${availableBuildToolsVersions.join(", ")} and none of them satisfies buildToolsRange: ${buildToolsRange}`);
+			}
+
+			this.printMessage(`You have to install Android Build Tools and the version should be in the following range: '${buildToolsRange}'.`,
+				'Run "android" from your command-line to install required Android Build Tools.');
```

Comment:
Perhaps "Run Android SDK Manager" instead of "Run android from your command-line".